objectiveC-day7.txt

```diff
@@ -0,0 +1,195 @@
+
+1)
+Counting the number of places where the object reference is used is called reference counting.
+When an object is created, the initial reference count is set to 1,Each time you need to ensure that the object be kept around, you effectively create a
+reference to it by incrementing its reference count by 1.
+
+Before Arc was introduced,programmers used 2 methods.
+a) automatic garbage collection:
+
+    With garbage collection, the system automatically determines which objects own which
+    other objects, automatically freeing up (or garbage-collecting) objects that are no longer referenced as space is needed during the program’s execution.
+ 
+b) manual reference counting:
+
+     If you do not want to depend on ARC (automatic reference counting),then you need to know how to deal with reference counts.
+     [object retain]  To increment  a reference count by 1.
+     [object release]  to decrement the reference count by 1.
+     When the reference count reaches 0,the system knows the object is no longer being used(not referenced anywhere),so it frees up memory
+     by sending the object  a dealloc message.
+
+
+
+Memory management using automatic reference counting:
+Arc was introduced in ios version 4.
+System determines when to retain an object,release or when to autorelease.Under the
```

Comment:
When user create reference of an object, then system increases the retain count.